### ISSUE TITLE
Undefined offset: 0" in phpoffice/phpexcel/Classes/PHPExcel/Reader/Excel5.php

### ISSUE THREAD

**Issue by denndk** [PHPOffice/PHPExcel#441]
Hi,

First of all thanks for a great package, works great expect for the issue I am having now. 

I have downloaded a generated Excel xls file from another site. If I try to load the file i get this error "Undefined offset: 0" in phpoffice/phpexcel/Classes/PHPExcel/Reader/Excel5.php at line 3679 which is around:

```
} else {
                $cell = $this->_phpSheet->getCell($columnString . ($row + 1));
                $cell->setValueExplicit($this->_sst[$index]['value'], PHPExcel_Cell_DataType::TYPE_STRING);
            }

            if (!$this->_readDataOnly) {
                // add style information
                $cell->setXfIndex($this->_mapCellXfIndex[$xfIndex]); <-- this line
            }
        }
```

This is my code (using maatwebsite laravel-excel package):

```
$reader = Excel::load(Input::get('file'), function($reader) {})->all()->toArray();
```

If I open the same file in Excel and save it again in the same format (XLS) and then try to load it, it works fine. 

It must be something with the original file, but what is wrong? I cannot change how the file is generated in the first place. 

Working file: http://denni.dk/files/works.xls
Non working file: http://denni.dk/files/not_working.xls

**Comment by jalind** [comment]
Yes, I now encounter this error with an Excel file generated for us by ADP and which we import. Apparently the default cell style in the XF record is not pointed to correctly and the subscript is invalid. In our case, it is zero when it should be 15. It is my understanding that the workbook/worksheet default is at fixed index key 15. My temporary fix in our library distribution is to check to see if the xfIndex derived from offsets 6 and 7 in the record data actually exists as an key in the _mapCellXfIndex array and, if not, then set xfIndex to 15 since that key should always exist. 

As a bug fix, this would probably be a good idea to use the default workbook/worksheet formatting (xfIndex set to 15) as a last resort if the XF record doesn't contain a usable index into the _mapCellXfIndex array.

**Comment by jalind** [comment]
I have a patch for this in 1.8.0:

The Excel spreadsheets affected have no apparent problem in Excel.

modify the _readColInfo() method in Excel5 class between the offset: 6 and offset: 8 code to use index 15 if the XF index doesn't exist (is not defined):

			// offset: 6; size: 2; index to XF record for default column formatting
			$xfIndex = self::_GetInt2d($recordData, 6);
            // if the default format index doesn't exist in the XF record, the worksheet/workbook default format
            // at fixed index 15 should always exist and can be used as the default cell format as a last resort
            if (!isset($this->_mapCellXfIndex[$xfIndex]))
            {
                $xfIndex = 15;
            }

			// offset: 8; size: 2; option flags

				// bit: 0; mask: 0x0001; 1= columns are hidden
				$isHidden = (0x0001 & self::_GetInt2d($recordData, 8)) >> 0;

				// bit: 10-8; mask: 0x0700; outline level of the columns (0 = no outline)
				$level = (0x0700 & self::_GetInt2d($recordData, 8)) >> 8;

				// bit: 12; mask: 0x1000; 1 = collapsed
				$isCollapsed = (0x1000 & self::_GetInt2d($recordData, 8)) >> 12;


which will use a higher-level default as a last resort.

**Comment by KoIIIeY** [comment]
ErrorException in Excel5.php line 3555:
Undefined offset: 15
